### PR TITLE
Buddy interaction

### PR DESCRIPTION
**Pull request by brickman1444**
Fully splits resource count / hunger from happiness.
Hunger is changed immediately when given resources.
Happiness changes every night depending on hunger.
Buddies sing a song based on happiness.
Buddies emote based on hunger after given resources.
Buddies emote over time both hunger and happiness.
Player stats are proportional to happiness and are updated every night.

### CODE REVIEW

**Comment by randomPoison** [comment]
Please give a more complete description of what you've changed.

File: Assets/Scripts/Buddy/BuddyStats.cs [hunk 1]
```diff
@@ -3,17 +3,68 @@
...
+
+	[ReadOnly("Current Happiness State")]
+	[SerializeField] AudioSource _currentHappinessSound = null;
+	[ReadOnly("Current Happiness Sound")]
+	[SerializeField] HappinessState _currentHappinessState = HappinessState.Invalid;
+
+	[SerializeField] Material _sadMaterial     = null;
+	[SerializeField] Material _happyMaterial   = null;
+	[SerializeField] Material _neutralMaterial = null;
+	[Tooltip("Time in seconds between happiness emotes")]
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyStats.cs]: Why is this tooltip on `_hungryMaterial`?

File: Assets/Scripts/Buddy/BuddyStats.cs [hunk 2]
```diff
@@ -61,6 +112,9 @@ void Awake()
 		_particlesRenderer = _particles.GetComponent<Renderer>();
 		owner = GameObject.FindObjectOfType<GodTag>();
 		BuddyManager.RegisterBuddy( this );
+		_happiness = _startingHappiness;
+		AdjustHappiness(0); // To initial sound and stuff
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyStats.cs]: Pad with spaces and typo in the comment.

File: Assets/Scripts/Buddy/BuddyStats.cs [hunk 3]
```diff
@@ -83,20 +137,40 @@ public void GiveResource( PlayerStats actorStats, ResourceData resourceData)
 	{
 		DebugUtils.Assert( _isAlive, "Cannot give a dead buddy resources." );
 
+		if ( resourceData == null )
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyStats.cs]: `resourceData` should never be null by the time it gets to this point so this should be an assert.

**Comment by randomPoison** [comment]
I'm done looking this over.

File: Assets/Scripts/Buddy/BuddyStats.cs (hunk 1 above)
Comment (brickman1444) [inline Assets/Scripts/Buddy/BuddyStats.cs]: fixed

File: Assets/Scripts/Buddy/BuddyStats.cs (hunk 2 above)
Comment (brickman1444) [inline Assets/Scripts/Buddy/BuddyStats.cs]: fixed

File: Assets/Scripts/Buddy/BuddyStats.cs (hunk 3 above)
Comment (brickman1444) [inline Assets/Scripts/Buddy/BuddyStats.cs]: fixed

**Pull request merged**